src/harness/fourslash.ts

```diff
@@ -1684,6 +1697,89 @@ namespace FourSlash {
             assert.equal(actualDefinitionContainerName, expectedContainerName, this.messageAtLastKnownMarker("Definition Info Container Name"));
         }
 
+        public goToImplementation(implIndex?: number) {
+            const implementations = this.languageService.getImplementationAtPosition(this.activeFile.fileName, this.currentCaretPosition);
+            if (!implementations || !implementations.length) {
+                this.raiseError("goToImplementation failed - expected to at least one implementation location but got 0");
+            }
+
+            if (implIndex === undefined && implementations.length > 1) {
+                this.raiseError(`goToImplementation failed - no index given but more than 1 implementation returned (${implementations.length})`);
+            }
+
+            if (implIndex >= implementations.length) {
+                this.raiseError(`goToImplementation failed - implIndex value (${implIndex}) exceeds implementation list size (${implementations.length})`);
+            }
+
+            const implementation = implementations[implIndex || 0];
+            this.openFile(implementation.fileName);
+            this.currentCaretPosition = implementation.textSpan.start;
+        }
+
+        public verifyRangesInImplementationList() {
+            const implementations: ImplementationLocationInformation[] = this.languageService.getImplementationAtPosition(this.activeFile.fileName, this.currentCaretPosition);
+            if (!implementations || !implementations.length) {
+                this.raiseError("verifyRangesInImplementationList failed - expected to at least one implementation location but got 0");
+            }
+
+            for (let i = 0; i < implementations.length; i++) {

```

Comment:
I wanted to be explicit about the error. Otherwise it would look like that particular implementation was not returned when actually it was returned twice.